plugins/common/shim/goshim.go

```diff
@@ -56,11 +59,13 @@ type Shim struct {
 // New creates a new shim interface
 func New() *Shim {
 	return &Shim{
-		metricCh: make(chan telegraf.Metric, 1),
-		stdin:    os.Stdin,
-		stdout:   os.Stdout,
-		stderr:   os.Stderr,
-		log:      logger.New("", "", ""),
+		BatchSize:    1,
+		BatchTimeout: 5 * time.Second,
```

Comment:
Defaults are usually 10s in telegraf
```suggestion
		BatchTimeout: 10 * time.Second,
```